flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/ClosableBlockingQueue.java

```diff
@@ -176,17 +178,17 @@ public boolean close() {
 	public boolean addIfOpen(E element) {
 		requireNonNull(element);
 
-		lock.lock();
+		writeLock.lock();
 		try {
 			if (open) {
 				elements.addLast(element);
 				if (elements.size() == 1) {
-					nonEmpty.signalAll();
+					nonEmpty.signal();
```

Comment:
`signal()` is safe and reasonable, since all the threads which waiting on the `nonEmpty` Condition wish to take element from queue, and at most one thread can get element from queue.